src/Kernel/Context.class.st

```diff
@@ -1772,6 +1772,31 @@ Context >> runUntilErrorOrReturnFrom: aSender [
 	].
 ]
 
+{ #category : #private }
+Context >> runUnwindTo: aContext onBehalfOf: aProcess [
+	"Run self on behalf of aProcess on aProcess's (i.e. self's) stack until aContext returns. Avoid a context that cannot return. Note: self must be a stack top context. Note: to avoid infinite recursion of MNU error: e.g. a wrapper around the message sentTo: receiver in #doesNotUnderstand: must be implemented.
+	This method is meant to be used exclusively by Process>>#terminate."
+
+	Processor activeProcess
+		evaluate: [ 
+			| here unwindBottom newTop |
```

Comment:
I would slit this method on two parts: 
- Context>>runUnwindsTo: with this code inside #evaluate (or similar).
- And Process>>#runUnwindsBetween:to: which would wrap "aContext runUnwindsTo:" with #evaluate:onBehalfOf:. See other senders of #evaluate:onBehalfOf: . They are all on the Process side.

Reply:
yes, I was considering this and checking typical senders of #evaluate:onBehalfOf: but then I decided against; no "big" reason though: Context it just seemed to me a better place for a method manipulating context similarly as #runUntilErrorOrReturnFrom: . #runUnwindTo:onBehalfOf: was conceived by pruning #runUntilErrorOrReturnFrom: as much as possible :) But you're right, it could as well be under Process...

Reply:
one of the reasons I put #runUnwind:onBehalfOf: under Context is I can emphasize the fact it's evaluated "on behalf of" in the name of the method; and other could be the process being terminated is the patient being operated on, so making it a receiver (the one object "responsible" for the action in the OO philosophy) didn't seem right :)